code/modules/games/cards.dm

```diff
@@ -102,51 +143,68 @@
 	var/list/to_discard = list()
 	for(var/datum/playingcard/P in cards)
 		to_discard[P.name] = P
-	var/discarding = tgui_input_list(user, "Which card do you wish to draw?", "Deck of Cards", to_discard)
+	var/discarding = tgui_input_list(user, "Which card do you wish to pick?", "Deck of Cards", to_discard)
 	if(!discarding || !to_discard[discarding] || !user || !src)
 		return
 
 	var/datum/playingcard/P = to_discard[discarding]
 	H.cards += P
 	cards -= P
 	H.update_icon()
-	user.visible_message("<b>\The [user]</b> draws a card.", SPAN_NOTICE("You draw the [P.name]."))
+	balloon_alert_to_viewers("<b>\The [user]</b> picks out a card.")
+	if(!length(cards))
+		qdel(src)
 
-/obj/item/deck/verb/deal_card()
+/obj/item/deck/verb/dealcard()
 	set category = "Object"
 	set name = "Deal"
-	set desc = "Deal a card from a deck."
+	set desc = "Deal a card from the deck."
 	set src in view(1)
 
-	if(usr.stat || !Adjacent(usr))
+	deal_card(usr, FALSE)
+
+/obj/item/deck/proc/deal_card(mob/user, mob/living/target)
+	if(use_check_and_message(user, USE_DISALLOW_SILICONS) || !Adjacent(user))
+		return
+	if(!iscarbon(user))
+		to_chat(user, SPAN_WARNING("Your simple form can't operate \the [src]."))
+	if(!length(cards))
+		to_chat(usr, SPAN_WARNING("There are no cards in \the [src]."))
 		return
 
-	if(!cards.len)
-		to_chat(usr, SPAN_WARNING("There are no cards in the deck."))
+	if(!target)
+		var/list/players = list()
+		for(var/mob/living/player in viewers(3))
+			if(!player.stat)
+				players += player
+
+		target = tgui_input_list(usr, "Who do you wish to deal a card?", "Deal", players)
+
+	if(target == user)
+		draw_card(user)
 		return
 
-	var/list/players = list()
-	for(var/mob/living/player in viewers(3))
-		if(!player.stat)
-			players += player
+	if(istype(get_step(target,target.dir), /obj/machinery/door/window) || istype(get_step(target,target.dir), (/obj/structure/window)))
+		return // should stop you from dragging through windows
 
-	var/mob/living/M = tgui_input_list(usr, "Who do you wish to deal a card?", "Deal", players)
-	if(!usr || !src || !M) return
+	if(!length(cards))
+		to_chat(user, SPAN_WARNING("There are no cards in the deck."))
+		return
 
-	deal_at(usr, M)
+	var/obj/item/hand/H = new hand_type(user.loc)
 
-/obj/item/deck/proc/deal_at(mob/user, mob/target)
-	var/obj/item/hand/H = new(get_step(user, user.dir))
+	H.randpixel = 6
+	H.randpixel_xy()
 
 	H.cards += cards[1]
 	cards -= cards[1]
-	H.concealed = 1
+	H.concealed = TRUE
 	H.update_icon()
-	if(user==target)
-		user.visible_message("<b>\The [user]</b> deals a card to [user.get_pronoun("himself")].")
-	else
-		user.visible_message("<b>\The [user]</b> deals a card to \the [target].")
-	H.throw_at(get_step(target,target.dir),10,1,H)
+	user.do_attack_animation(src, null)
```

Comment:
not particularly sure, i think this looks odd

Reply:
it's in line with the normal throwing animations